src/aics-image-viewer/components/ChannelsWidgetRow/index.tsx

```diff
@@ -0,0 +1,270 @@
+
+const ISOSURFACE_OPACITY_DEFAULT = 1.0;
+const ISOVALUE_DEFAULT = 128.0;
+
+interface ChannelsWidgetRowProps {
+  index: number;
+  imageName: string;
+  channelName: string;
+  name: string;
+  volumeChecked: boolean;
+  isosurfaceChecked: boolean;
+  colorizeEnabled: boolean;
+  colorizeAlpha: number;
+  color: [number, number, number];
+  channelControlPoints: {
+    color: string;
+    opacity: number;
+    x: number;
+  }[];
+  channelDataForChannel: any; // volume-viewer Channel type
+
+  changeOneChannelSetting: (channelName: string, channelIndex: number, keyToChange: string, newValue: any) => void;
+  handleChangeToImage: (keyToChange: string, newValue: any, index?: number) => void;
+  updateChannelTransferFunction: (index: number, lut: Uint8Array) => void;
+  onColorChangeComplete?: (newRGB: ColorObject, oldRGB?: ColorObject, index?: number) => void;
+}
+
+export default class ChannelsWidgetRow extends React.Component<ChannelsWidgetRowProps, { controlsOpen: boolean }> {
+  constructor(props: ChannelsWidgetRowProps) {
+    super(props);
+    this.toggleControlsOpen = this.toggleControlsOpen.bind(this);
+    this.onColorChange = this.onColorChange.bind(this);
+    this.volumeCheckHandler = this.volumeCheckHandler.bind(this);
+    this.isosurfaceCheckHandler = this.isosurfaceCheckHandler.bind(this);
+    this.state = {
+      controlsOpen: false,
+    };
+  }
+
+  volumeCheckHandler({ target }: { target: { checked: boolean } }) {
+    const { channelName, index, changeOneChannelSetting, isosurfaceChecked } = this.props;
+    if (!target.checked && !isosurfaceChecked) {
+      this.setState({ controlsOpen: false });
+    }
+    changeOneChannelSetting(channelName, index, VOLUME_ENABLED, target.checked);
+  }
+
+  isosurfaceCheckHandler({ target }: { target: { checked: boolean } }) {
+    const { channelName, index, changeOneChannelSetting, volumeChecked } = this.props;
+    if (!target.checked && !volumeChecked) {
+      this.setState({ controlsOpen: false });
+    }
+    changeOneChannelSetting(channelName, index, ISO_SURFACE_ENABLED, target.checked);
+  }
+
+  createChannelSettingHandler = (settingKey: string) => (newValue: any) => {
+    const { channelName, index, changeOneChannelSetting } = this.props;
+    changeOneChannelSetting(channelName, index, settingKey, newValue);
+  };
+
+  onIsovalueChange = this.createChannelSettingHandler(ISO_VALUE);
+  onOpacityChangeUnwrapped = this.createChannelSettingHandler(OPACITY);
+  onOpacityChange = (newValue: number) => this.onOpacityChangeUnwrapped(newValue / ISOSURFACE_OPACITY_SLIDER_MAX);
+
+  createSliderRow = (name: string, maxValue: number, defaultValue: number, onChange: (newValue: any) => void) => (
+    <Row>
+      <Col span={10}>
+        <label style={STYLES.controlName}>{name}</label>
+      </Col>
+      <Col span={12}>
+        <Slider // TODO: this is the only remaining place we're using antd's slider rather than Nouislider. Replace it?
```

Comment:
interesting possible problem: the react wrapper we use for nouislider is currently way behind nouislider itself :(

Reply:
the number one reason we use nouislider is for the "range slider" with two handles, and the convenient ability to drag the section between the handles to move it.

Reply:
I primarily bring it up for design consistency. It's odd to have this little buried corner of the UI use a slider that looks completely different from the rest of the sliders in the app.

Reply:
agreed, it's good to have that comment reminder in there.